server/routers/activity.ts

```diff
@@ -27,6 +48,7 @@ export const activityRouter = createTRPCRouter({
       const activity = await ctx.prisma.activity.findUnique({
         where: {
           id: input.id,
+          userId: ctx.user?.id,
```

Comment:
This is not needed. We should query by id only and if `ctx.user.id !== activity.userId` we should return 403. Because it would mean user is trying to access someone's else activity